### PR TITLE
Fixed Mantis #18387 

### PR DESCRIPTION
**Pull request by fneumann**
Fixed Mantis #18387 ("Refresh Languages" of a plugin will always replace local changes"). 
This fix can be cherry-picked for ILIAS 5.0 and 5.1, too (ilObjLanguage is alredy prepared there).

The function ilPlugin::updateLanguages() gets a new parameter with the languages that should be updated. This will help to restrict the automatic update of plugins if only selected languages are updated in the language administration. If the parameter is null, then the plugin will be updated for all installed and supported languages. Local changes are queried for the module of the plugin and will not be replaced, if they exist.

### CODE REVIEW

File: Services/Component/classes/class.ilPlugin.php [hunk 1]
```diff
@@ -394,23 +394,45 @@ function getTablePrefix()
...
 		ilGlobalCache::flushAll();
 		include_once("./Services/Language/classes/class.ilObjLanguage.php");
 
+		// get the keys of all installed languages if keys are not provided
+		if(!isset($a_lang_keys))
+		{
+			$a_lang_keys = array();
+			foreach (ilObjLanguage::getInstalledLanguages() as $langObj)
+			{
+				if ($langObj->isInstalled())
```
Comment (chfsx) [inline Services/Component/classes/class.ilPlugin.php:411]: Isn't this a bit redudant, or are there some situations where getInstalledLanguages() return languages, which are not installed?

**Review by chfsx — COMMENTED** [review]
Thanks @fneumann for the PR, I'm not sure if I'm the right person to accept it, because I'm just second-maintainer. But from my point of view this seems to be ok, the comment made in line 411 is more a question...

File: Services/Component/classes/class.ilPlugin.php (hunk 1 above)
Comment (fneumann) [reply Services/Component/classes/class.ilPlugin.php:411]: The check isInstalled() ins not needed, indeed. It's a relict before I added getInstalledLanguages to ilObjLanguage.

**Comment by alex40724** [comment]
@fneumann Thanks!

**Comment by PurHur** [comment]
@fneumann Thanks a lot! I will update a hell of ilias installations because of this bug

**Pull request merged**